### ISSUE TITLE
Random black screens

### ISSUE THREAD

**Issue by AlvaroBrey** [z2-dev/android_device_zuk_z2_plus#21]
Random black screens with white led on are reported on most nougat ROMs for this device. 
Edit: ramoops has been enabled, issue seems related to modem

**Comment by marchi19217** [comment]
This has happened to me on other ZUK ROMs but not this one thus far. I did make sure I erased all storage including SYSTEM when flashing this ROM

**Comment by luci-11** [comment]
any news about that? did you manage to catch a log?

**Comment by marchi19217** [comment]
Can confirm this is present on my device
[bugreport-NMF26V-2017-02-25-10-34-27.zip](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/800715/bugreport-NMF26V-2017-02-25-10-34-27.zip)

**Comment by koichirose** [comment]
Managed to get a console-ramoops thanks to the latest build (20170224). The file was stored in /sys/fs/pstore.
Attaching it here, hope it's useful!
[console-ramoops-koichirose-20171227.txt](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/803049/console-ramoops-koichirose-20171227.txt)

**Comment by koichirose** [comment]
Happened again after a couple of days.
Error is the same (something about SLPI), this time it happened right after I turned on the screen.

Can I help out in some other way?
[console-ramoops-koichirose-20170302.txt](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/814160/console-ramoops-koichirose-20170302.txt)

**Comment by AlvaroBrey** [comment]
Thanks @koichirose , the fact that it's the same error is good, means that it's not a coincidence.
SLPI is apparently related to the modem but the information in the ramoops is pretty limited. I'm still investigating.

**Comment by fraCPH** [comment]
I can easliy reproduce the bug, I also posted an adb bugreport on the zukfans forum.
Let me know if I can be of any help with a log, and how to get it. I'm with stock ZUI and not rooted though.

**Comment by AlvaroBrey** [comment]
@fraCPH It would be interesting to know _how_ you can reproduce the bug
Edit: also, where did you post the bugreport? Link?

**Comment by fraCPH** [comment]
These posts:
http://zukfans.eu/community/threads/zuk-z2-multiple-roms-freeze-with-white-led-on.3514/page-5#post-30742
http://zukfans.eu/community/threads/zuk-z2-multiple-roms-freeze-with-white-led-on.3514/page-6#post-31332
My phone never freezes if I keep it in airplane mode (whether wifi is ON or OFF), but as soon as I switch off the airplane mode it's a matter of seconds/minutes and it freezes. This has happened once with a SIM card in place and several other times with no SIM card in.

**Comment by poseidonss** [comment]
I can reproduce the bug easely also. What can I do to help you guys?

**Comment by fr0x3n** [comment]
Hi ontherunvaro,
I'm not an android dev but I'm facing the same issue (freeze black screen with notification led on). If I can be of any help, please let me know.

Also considering that we have to force a reboot (long press power button) I don't know how to extract a log that contain relevant information. Would you be so kind to explain how so I or any other can provide you with multiple log?

I've run different test here :
    - never freeze while connected
    - never freeze while screen on (at least for me)
    - freeze always occured while phone locked
    - I've been running "screen backlight off" instead of locking the phone and it never freeze since (about a month) except while I accidentally lock.

**Comment by AlvaroBrey** [comment]
@poseidonss , @fr0x3n 
Logs after a kernel panic can only be found in console-ramoops:
`adb shell cat /sys/fs/pstore/console-ramoops > ramoops.txt` on your computer, on the first reboot after the black screen.  Obviously you need  `adb` installed on your pc, and usb debugging enabled on the phone.

**Comment by koichirose** [comment]
@poseidonss, @fr0x3n you can also do it directly on the phone as I did, simply open a root browser, copy /sys/fs/pstore/console-ramoops somewhere in local storage and then post it here.

**Comment by fr0x3n** [comment]
well I tried but I don't have any "pstore" folder in /sys/fs/
I also tried to search for a "console-ramoops" file in / but nothing came up (except "console-ramoops-koichirose-20171227.txt" from my download folder)

Is there any other command to launch prior to that in order to activate monitoring?

**Comment by koichirose** [comment]
@fr0x3n you need to be at least on build 20170224, it was enabled in that build.
https://github.com/ontherunvaro/android_device_zuk_z2_plus/wiki/Changelog

**Comment by fr0x3n** [comment]
Ahhh thats why... Sorry for my noob question but I'm current on Dirty_Unicorn_z2_plus_7.1.1_20170206-2233.v11.0-UNOFFICIAL

What rom do I need to flash ? Or maybe I can stay with this rom and flash something else?

**Comment by AlvaroBrey** [comment]
@fr0x3n issues in this repository are just for my LineageOS builds, not for any other ROM.

**Comment by fr0x3n** [comment]
ok my mistake again though this issue is general on any rom i've tried.
I'm downloading your rom and keep you posted.

Thanks

**Comment by poseidonss** [comment]
[ramoops.txt](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/835962/ramoops.txt)
[ramoops1.txt](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/835963/ramoops1.txt)

Here are the ramoops files.

Its very easy to reproduce so I did it twice.

**Comment by poseidonss** [comment]
[ramoops_lineageos.txt](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/836014/ramoops_lineageos.txt)
These two I posted was with lastest stock chinese rom.
This one is with lineageos.

They all happen the same way:
Right after I lock the screen.

**Comment by ontherunvaro** [comment]
@poseidonss what is your baseband version?

**Comment by poseidonss** [comment]
M8996FAAAANAZM-1.70664.3.76387.1

2017-03-11 19:23 GMT-03:00 Álvaro Brey <notifications@github.com>:

**Comment by poseidonss** [comment]
2017-03-11 19:23 GMT-03:00 Álvaro Brey <notifications@github.com>:

**Comment by SuperLamic** [comment]
@poseidonss what are the steps to reproduce it? I'm running my phone for a week on Lineage and I haven't had any reboot/black screen.

**Comment by poseidonss** [comment]
@SuperLamic I just need to lock the screen.

**Comment by SuperLamic** [comment]
@poseidonss have you tried reflashing the phone via qfil (zui 2.5) and flashing Lineage back?

**Comment by poseidonss** [comment]
@SuperLamic Yes, I think I flashed every rom there is. No success. The only thing that keeps the phone working is a wakelock app.

**Comment by marchi19217** [comment]
Did you try formatting all partitions including system/cache/data? Also try after a clean install with no third party apps

Seems you have hardware issues. Regardless it's not specific to this ROM

After 2 months of using this rom I have experienced the black screen once

**Comment by fraCPH** [comment]
> Seems you have hardware issues. Regardless it's not specific to this ROM

That's not very encouraging.
@ontherunvaro are there any comments about the ramoops posted by @poseidonss ?
Is it worth that someone else tries installing your ROM to produce other bug reports, or, given that this is (likely) a baseband problem, you cannot do anything about that ?
In this case I would keep the stock ZUI to maintain the warranty and send the phone back.

**Comment by marchi19217** [comment]
Okay, while trying to reproduce the "Network connection error" bug I had the black screen error happen again. Opened the SIM tray, inserted a second SIM, reinserted tray, phone crashed

**Comment by kpcz** [comment]
Saddly already had 3 times freeze since friday (lastest release) 

[ramoops.txt](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/871008/ramoops.txt)

Is there any possibility to fix this, or looks like it is hardware failure of the device? On other ROMs had the same issue.

Baseband version M8996FAAAANAZM-1.70664.3.76387.1

**Comment by AlvaroBrey** [comment]
All the info we have right now are these lines:
```
[ 7540.170827] Fatal error on slpi!
[ 7540.170878] slpi subsystem failure reason: PID 1 :  .
[ 7540.274889] Kernel panic - not syncing: subsys-restart: Resetting the SoC - slpi crashed.
```
These lines appear on almost every crash log we've captured or been sent.
We know it's related to modem/baseband but we frankly don't have any more info for now.

**Comment by AlvaroBrey** [comment]
Also, the phone is supposed to reset (`Resetting the SoC`), not to hang on a black screen, so there's more than just 1 problem there.

**Comment by kpcz** [comment]
Is it somehow connected to baseband version? Maybe i can try the 2.0.c1.9-00026-M8996FAAAANAZM-1 baseband, or does the issue is present on both of them? Thanks.

**Comment by AlvaroBrey** [comment]
@kpcz since you have so many black screens, _you_ should try changing basebands and watching out for any changes in frequency of black screens.
I personally can't test this because I've had maybe 3 black screens in the 3 months I've had the device.

**Comment by marchi19217** [comment]
I'm on the same baseband and rarely see a black screen, I think the black screen/solid led is more a general consequence of a fatal error, most other devices would just hard reboot when such an error occurred

**Comment by digitalpriory** [comment]
Just thought I'd chime in and say I've had all of the above issues; after recently bricking my phone (unrelated reasons), I wanted to try this ROM after using Harsha's ROM religiously.  

After a factory reset and without storing any backups, I get the black screens (accompanied with a crash in the start wizard, that I'll report separately).  

All these issues seem to be inapplicable when switching back to Harsha's ROM.  Will upload logs later but thought it was worth corroborating other findings here.

**Comment by marchi19217** [comment]
@digitalpriory Did you erase all partitions except for sdcard? This could potentially explain the system crash/black screen in your specific case which does not seem to be a hardware fault

**Comment by AlvaroBrey** [comment]
@digitalpriory I don't think it's a likely thing that Harsha's ROM didn't have any black screens, because they are present even in stock roms. Specially since harsha's builds were just compiled off of Davide's and Fedor's sources without any further modifications. You might have just lucked out until now.
Either way, your logs will be appreciated, and all info is welcome.

**Comment by AlvaroBrey** [comment]
By the way, I've recently made a commit to kernel (https://github.com/ontherunvaro/android_kernel_zuk_msm8996/commit/ba7db4837bcc792e3858f480cd27c058988ad819) that will make the phone restart on kernel panics, instead of hanging on the black screen. This should hopefully alleviate the issue, but we still have to look for the cause.

**Comment by koichirose** [comment]
Thank you @ontherunvaro, looking forward to testing it.
Restarting would definitely alleviate the issue. I experienced just one black screen with your latest build, whereas with previous builds I had at least one per day.
Just wanted to point out again that I don't think this is necessarily hardware related. I never had a black screen in at least three weeks with @davidevinavil's RR and some builds of harsha.pic's Lineage.

In the meantime, I tried writing to the ZUK support without any great hopes of getting an answer. Let's see.

**Comment by kpcz** [comment]
Thanks @ontherunvaro , these are really great news, as it would be great workaround for the problem for now 👍  @koichirose yes with harsha.pic's Lineage I had less black screens as well. But not depending on ROMs I had it with all possible ROMs I have tried. Please share the link where you can write to ZUK support, I will write as well. Already wrote on Lenovo forums, no response. Also on http://bbs.lenovomobile.cn lot of people are complaining about this issue.

**Comment by AlvaroBrey** [comment]
@koichirose @kpcz I can send you a test build to see if it's working properly. Send me an email to varo@ontherunvaro.com or speak to me in Telegram at @ontherunvaro and I will reply to you with a link.

**Comment by digitalpriory** [comment]
@marchi19217 well, everything is deleted in TWRP including sdcard partition.  What seemed to work was reverting to the pre-hotfix release, which let me get in and reinstall all apps but I found Button settings to break/crash.  Then I dirty flashed the hotfix release over it and this seems to now be stable.  Just FYI - U-Touch appears to be missing the 'Back' option on my installation; not sure if this is intended, known or just specific to what's going on at my end.  Thanks.

**Comment by digitalpriory** [comment]
I really must applaud what you guys are doing since no one else is doing this for LineageOS on this phone, but I've just tested the camera back against Harsha's implementation of the Snap camera and I'm switching back to one of my backups on that ROM until a later date.

**Comment by AlvaroBrey** [comment]
@digitalpriory Hotfix release's only difference to the previous one is literally just the camera crash fix, the rest is exactly the same.
Back (short tap) and Home (press) cannot have their functions customized, that's a LineageOS limitation.

**Comment by digitalpriory** [comment]
@ontherunvaro - hey yeah, I appreciate that's the case as I read the changelog and it makes no sense to me either why there were random issues (for me) in the hotfix.  

The Button settings itself was an issue on its own before the 'back' setting as it refused to load the screen at all on the pre-hotfix release, but then worked fine once I dirty flashed.

Ultimately, this ROM is very close for me as I can alter the keylayout files (which I needed to do with Harsha's ROM too, so that isn't a dealbreaker, but the camera issue still isn't quite right even after installing several other camera apps.  I'll definitely end up here eventually...  I know we're digressing a bit, and I'm sure you've done this already, but I'd definitely suggest taking a look at the Snap integration on Harsha's ROM.  It's spot on, superb quality, but slow mo doesn't work.  I have no idea what to suggest regarding the black screens but what I will say is that I've never experienced any of them on the Harsha's.  Sorry, I sound like a broken record...

**Comment by AlvaroBrey** [comment]
@digitalpriory what I get from your comments on camera is that it takes better pictures/videos? in what ways? I'm not a heavy camera user, so camera in my builds seems alright to me

**Comment by koichirose** [comment]
@kpcz I simply contacted them here: http://www.zukmobile.cc/contacts/index/
@ontherunvaro I contacted you on telegram :)

**Comment by koichirose** [comment]
Just experienced my first reboot with @ontherunvaro's test build.
This is already a lot better, since rebooting takes seconds and it almost always happens when I'm not using the device.

**Comment by kpcz** [comment]
@koichirose great news. Mine ZUK haven't experienced kernel panic yet, so haven't seen any reboots (have widget with uptime). What baseband version do you use? As it looks like you experience freezes more often than me.

**Comment by koichirose** [comment]
I'm on 1.53476.1, what about you?
I experience one freeze (now reboot!) every 2-3 days.
Always wanted to update but then I wouldn't be able to help out with fixing this issue.

**Comment by kpcz** [comment]
i am on M8996FAAAANAZM-1.70664.3.76387.1, installed this ROM on Friday, and since I had about 3 freezes. As this is the best ROM I have tried, Im sticking with it.  (I tried lot of them)

**Comment by kpcz** [comment]
Just had a reboot, not a freeze, so I can confirm that the update works, and its already lot better now.

**Comment by fr0x3n** [comment]
Hi,
I flashed lineage-14.1-20170324-hotfix-UNOFFICIAL-z2_plus.zip yesterday and got two freezes here are the two console-ramoops files

[console-ramoops 2017.03.29-08.txt](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/878498/console-ramoops.2017.03.29-08.txt)
[console-ramoops 2017.03.28-19.txt](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/878497/console-ramoops.2017.03.28-19.txt)

Could you please explain how to find the crash in those files (though I won't be able to do anything about it :))

Have I flashed the wrong rom version ? I wanted the version with the fix to reboot instead of freeze. Or maybe the problem that i'm facing is somewhat different. It's a mystery to me.

**Comment by AlvaroBrey** [comment]
@fr0x3n The reboot fix isn't on any published rom yet, wait for release!
Either that or contact me to ask for test build link (see previous posts)

**Comment by fr0x3n** [comment]
Oh thats why !! Thanks
Any help with where in ramoops file, i can find crash info?

**Comment by kpcz** [comment]
@fr0x3n just few comments above.  https://github.com/ontherunvaro/android_device_zuk_z2_plus/issues/21#issuecomment-289311382 ;)

**Comment by fr0x3n** [comment]
oh sorry I missed that one. Thanks

strange though I search both files I attached above and none contain either "fatal error" nor "kernel panic". is it possible that the issue i'm facing here is somewhat different? could you please have a look?

**Comment by fr0x3n** [comment]
Looking into ramoops file transmitted by others I noticed that theres start at a certain timestamp "[ 5878.792490]" whereas mine both start at "[    0.000000]" did I missed something again (hard to be a total noob)
Im' using latest rom "lineage-14.1-20170324-hotfix-UNOFFICIAL-z2_plus.zip" available at https://www.androidfilehost.com/?w=files&flid=151872

**Comment by AlvaroBrey** [comment]
@fr0x3n I don't know how you're capturing those files, but you're supposed to get them after rebooting the phone from a black screen.
That is:

1. Black screen
2. Reboot by holding button
3. Let the phone boot completely
4. Get ramoops file.

It's important that it's done in this order and that there's no more reboots between the black screen and the ramoops file dump.

**Comment by fr0x3n** [comment]
@ontherunvaro Yes, that's precisely what I've done. I captured those file directly on the phone in /sys/fs/pstore/ using a file explorer and not "adb shell cat /sys/fs/pstore/console-ramoops > ramoops.txt".
I'll try the adb version when next freeze will occured (but it should be good to using a file explorer)

In the mean time could you have a look at those ramoops file a attached above to see if the problem I'm facing is the same as the one you identify? it seems so considering blackscreen, notification led flashing.

**Comment by AlvaroBrey** [comment]
@fr0x3n I looked over them earlier and coudln't find any info related to the crash, dunno why.

**Comment by fr0x3n** [comment]
ok i'll try using adb next time.

Is there a way for me to add a "patch" so that my phone reboot instead of freeze or have I to wait for a rom update?

**Comment by AlvaroBrey** [comment]
@fr0x3n just wait, new version will be out later this week

**Comment by fraCPH** [comment]
I also can confirm that with the experimental build I get reboots rather than freezes. 
However in my case reboots turn into bootloops which I can stop only by entering the recovery and deleting cache and dalvik cache (and in one case I had to to it twice because the first time it wasn't enough), so I have also lost my ramoops. 
The phone for me remains totally useless :-(

**Comment by poseidonss** [comment]
@fraCPH have you tried using a wakelock app? its the only thing that keeps the my phone working,

**Comment by fraCPH** [comment]
@poseidonss if I understand correctly your phone crashes only when the screen is off, while mine crashes often also with the screen on, so I'm not sure a wakelock app could change the things.
However I will give it a try, can you suggest one ?

**Comment by AlvaroBrey** [comment]
@fraCPH Ugh, yours seems like a more serious issue than the typical modem crash we're seeing.
Getting one of your ramoops would be very interesting. You may be able to get them from recovery (`adb pull /sys/fs/pstore/console-ramoops`).
If you can't, please send any files inside `/data/tombstones` after a reboot.
Edit: also  `/data/anr/traces.txt` if it exists.

**Comment by fraCPH** [comment]
@ontherunvaro I have just triggered a reboot and, by stopping at the recovery, I was able to pull the tombstones and the traces.txt files, however there is no pstore directory in the /sys/fs/ path.
[Tombstones.zip](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/883554/Tombstones.zip)
[traces.txt](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/883548/traces.txt)

@SuperLamic Edit: after rebooting into the OS I do have the pstore directory but it's empty.
I also attach the bugreport captured during the crash.
[bugreportLineage.txt](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/883588/bugreportLineage.txt)

**Comment by SuperLamic** [comment]
@fraCPH you have to reboot to android to get ramoops

**Comment by AlvaroBrey** [comment]
Right, there will only be a pstore in recovery if the TWRP was build with a kernel after I activated it (like my TWRP builds for example) as far as I know.

**Comment by flythq** [comment]
Somebody tried the radio M8996FAAAANAZM-1.63821.1 which is from the Indian 2.0.133 ST? It seems that with this modem I have't received a death screen for more than a week. Before that there were 2-3 per day

**Comment by SuperLamic** [comment]
@poseidonss @fraCPH can you guys try issuing this command in terminal (with su)? It could fix your issue.

`setprop persist.sys.ssr.restart_level venus,AR6320,slpi,modem,adsp`

_Edit from repo owner: formatting_

**Comment by poseidonss** [comment]
What does it do?

Em 4 de abr de 2017 2:35 PM, "SuperLamic" <notifications@github.com>
escreveu:

**Comment by SuperLamic** [comment]
@poseidonss it sets slpi (that causes panic) to lower restart level, so only slpi subsystem should restart instead of whole phone. You could set this prop in build.prop to set it permanently. This will last only to next reboot.

**Comment by poseidonss** [comment]
Sounds good :D

Em 4 de abr de 2017 2:39 PM, "SuperLamic" <notifications@github.com>
escreveu:

**Comment by kpcz** [comment]
@SuperLamic  command executed. Any way to check if it's applied correctly?

**Comment by SuperLamic** [comment]
@kpcz just that it won't crash.. (black screen or now reboot)

if you already have the build with reboots (and not black screens), just check uptime next day or something.

**Comment by SuperLamic** [comment]
or you can use `getprop persist.sys.ssr.restart_level` to check if it's set..

**Comment by kpcz** [comment]
@SuperLamic yes it's set correctly. Will check my uptime widget and report in few days. Thanks :)

**Comment by koichirose** [comment]
I did that, too and checked with getprop, will report back in a few days!
Thanks

**Comment by poseidonss** [comment]
Just runned it.

Locked the screen twice and no reboot for now.
Going to test more.

2017-04-04 15:10 GMT-03:00 koichirose <notifications@github.com>:

> I did that, too and checked with getprop, will report back in a few days!
> Thanks
>

**Comment by poseidonss** [comment]
Well, the wakelock was activated. Deactivated it and the phone booted.

--- >' <---- Is this part of the command? I can´t execute with it .setprop
persist.sys.ssr.restart_level venus,AR6320,slpi,modem,adsp

2017-04-04 16:17 GMT-03:00 Evandro Sanches <evandrosanches2112@gmail.com>:

> Just runned it.
>
> Locked the screen twice and no reboot for now.
> Going to test more.
>
> 2017-04-04 15:10 GMT-03:00 koichirose <notifications@github.com>:
>
>> I did that, too and checked with getprop, will report back in a few days!
>> Thanks
>>

**Comment by AlvaroBrey** [comment]
I will make a test build tomorrow, stay tuned.

**Comment by fraCPH** [comment]
I am doing it wrong. Better if I wait the test build...
![screenshot_20170404-225050](https://cloud.githubusercontent.com/assets/26164267/24678126/a641447a-1989-11e7-9786-f34d850231ce.png)

**Comment by SuperLamic** [comment]
@fraCPH I recommand you to use terminal emulator. Then issue only su and then the setprop.

**Comment by fraCPH** [comment]
Issuing su and setprop as separate commands worked.
I then tried to disable airplane mode and I instantly got a crash, so no better luck.
However this time the phone rebooted straight to the OS without bootloops and I managed to get a ramoops, maybe it helps.

[ramoopsLineage.txt](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/894559/ramoopsLineage.txt)

P.S.
I was using Termux because with Terminal emulator the keyboard in landscape mode doesn't work and in portrait mode you can't see the whole line if it's too long

**Comment by AlvaroBrey** [comment]
I have some bad news. I'm gonna have to revert the reboot workaround until I investigate some more, since it disables the emergency download mode. This potentially means you wouldn't be able to recover from a softbrick, and that is far more dangerous than black screens.

In the meantime I'm doing a test build with a change similar to what @SuperLamic proposed, I will upload it in a couple of hours.

**Comment by koichirose** [comment]
Hi, in the meantime I can report that I experienced a reboot today, after executing the command @SuperLamic suggested.

**Comment by AlvaroBrey** [comment]
@koichirose @fraCPH @poseidonss @kpcz Here's a test build:
https://mega.nz/#!edwCTA6A!i0Q-oogikq-IwR0aeqY1rP8j91xddS8gODg-HnEoPko
Tell me if you experience any changes in the frequency of crashes.
**Beware**: I've already disabled the reboot workaround for reasons exposed above, so if it crashes, you will get a black screen as before, not a reboot.
Edit: don't mind the silly bootanimation or the Br0Zip app, they are April Fools jokes from the LineageOS team.

**Comment by kpcz** [comment]
@ontherunvaro last time I updated (previous builds), somhow the gapps broke (all google accounts were missing from accounts section and could not add any anymore). 
When updating I need to flash ROM + gapps also, correct?

Its really sad news about reboot workaround.. :(

**Comment by AlvaroBrey** [comment]
@kpcz Yes, you need to flash GApps after flashing the ROM if you follow the instructions on my thread. 
However, please keep offtopic to a minimum, this is not a forum. That also means you don't need to reply to this comment with "sorry for offtopic" lol.

**Comment by fraCPH** [comment]
As expected nothing changes with today's version.
Here's the ramoops I collected after the freeze.

[console-ramoops 04-05.txt](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/900613/console-ramoops.04-05.txt)

And this one half an hour later
[console-ramoops 04-05a.txt](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/900761/console-ramoops.04-05a.txt)

**Comment by AlvaroBrey** [comment]
@fraCPH I don't know why but I can't find any information related to your crashes in the files you've posted. Maybe they were cut or something? In some of them I see the traces of the device booting but not of the crash, it's weird.

**Comment by SuperLamic** [comment]
@ontherunvaro seems like a hard reboot (via holding power button). Indeed I find it in most logs too. Maybe they don't understand that they have to grab the log RIGHT when their device first boots into android... I don't know, but these logs are without any error. (that would cause kernel panic)

**Comment by poseidonss** [comment]
I, rarely, can make this bug stop if I'm lucky(its totally random) flashing a ROM. The phone don't crash, even without wakelock app, until I reboot it then start crashing again. So there might be a way to stop this crash to happen.
I also can make it crashes even with the wakelock app, I just need to put it in airplane mode.

**Comment by koichirose** [comment]
What exactly is the risk of rebooting instead of showing the black screen?

**Comment by SuperLamic** [comment]
@koichirose you can't reboot into download mode at all (it reboots into it when crash happens)

**Comment by koichirose** [comment]
What exactly is download mode, when would I need it and why would it
happen?
Is it within TWRP, to adb push files?
Thanks

**Comment by SuperLamic** [comment]
@koichirose when you need to use qfil, you use download mode

**Comment by koichirose** [comment]
Got it. But is there a risk of me needing qfil to unbrick the device? What
would have to happen for me to need it and why didn't it happen until now?
I have been experiencing reboots daily since the release with reboots
instead of black screens came out.

Sorry about all these questions, just trying to understand a bit more.

**Comment by AlvaroBrey** [comment]
@koichirose I'll try to explain it neatly:

- EDL/download mode is a low-level utility that allows you to flash a ROM directly to the phone memory even if recovery/etc are broken. As such, it is invaluable in case of a brick because it has the potential to repair it.
- Black screens are happening because when the phone crashes, instead of rebooting (normal behaviour), it's entering download mode for some reason.
- In the previous build I disabled this behaviour, but in doing so I inadvertently disabled the possibility to enter download mode willingly.
- I am going to try to disable the black screens without disabling download mode altogether, just haven't gotten to that yet.

Hope that clears things up

**Comment by koichirose** [comment]
Super clear now, thank you!
What's still unclear is how would one brick the device and need download
mode when the rom is pretty stable and the only flashing-related stuff is
to update the rom itself.

I'll still update to the latest release asap.

**Comment by koichirose** [comment]
Just wanted to report that since I installed release 20140407 (5 days ago) I had zero black screens.
The changelog only mentions the reverted reboot workaround, did anything else change?

**Comment by AlvaroBrey** [comment]
@koichirose I'm happy to read that but unfortunately no, I didn't change anything else related to reboots on latest release.

**Comment by poseidonss** [comment]
Well, just turned off my wakelock app and no crash yet also.

2017-04-13 8:07 GMT-03:00 Álvaro Brey <notifications@github.com>:

**Comment by poseidonss** [comment]
It was my bank app that was running on backgroud. Closed it and crashed
again. =(

2017-04-13 10:01 GMT-03:00 Evandro Sanches <evandrosanches2112@gmail.com>:

**Comment by fraCPH** [comment]
@poseidonss can you name your wakelock app? I'd like to test it.

**Comment by poseidonss** [comment]
CPU awake

Em 13 de abr de 2017 5:18 PM, "fraCPH" <notifications@github.com> escreveu:

**Comment by fraCPH** [comment]
I tried it this morning and after 4 hours I believed I had solved the problem, but then I had another crash, and then another one after 1 hour.
In both cases the phone freezed again a couple of times during the reboot.
Here are the logs I collected as soon as I finally managed to boot into the OS.

[console-ramoops 17-04-14-01.txt](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/922497/console-ramoops.17-04-14-01.txt)
[console-ramoops 17-04-14-02.txt](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/922498/console-ramoops.17-04-14-02.txt)

**Comment by lastfreenick** [comment]
I've been using the rom for a few days now and all of a sudden the black screens started to show up two or three days ago which is strange because I had none in the beginning and then I realized after installing the rom I played around with the settings and set the networks of both simcards I use to prefer 3g networks because my provider does not offer 4g. Playing around in the settings again about three days ago I set both sim cards back to lte preferred. 
Today, after remembering that the black screens are baseband related, I set the simcards to 3g for my data card and 2g for my voice card and went from half a dozen black screens to none so far. 
Of course this could be a coincidence, but maybe someone with lots of black screens could test it out.

**Comment by zhadel2** [comment]
i'm using 20170414 rom without Gapps, without root,, black sceen with led on for me with ur latest rom: 
[ramoops.txt](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/923821/ramoops.txt)
[ramoops1.txt](https://github.com/ontherunvaro/android_device_zuk_z2_plus/files/923972/ramoops1.txt)

**Comment by zhadel2** [comment]
i don't have this problem with zui 2.3 oficial rom, all other, i get black screen

**Comment by AlvaroBrey** [comment]
Unfortunately seems the reboot workaround has stopped working on build 20170811

**Comment by AlvaroBrey** [comment]
Reboot workaround is back via https://github.com/z2-dev/android_kernel_zuk_msm8996/commit/62278446681504140c02a07b7f581cec3c457a92

**Comment by koichirose** [comment]
@ontherunvaro I don't have any issues on 20170811 about black screens/reboots.
I don't know if you remember, but I've had this issue in every build until this: https://github.com/z2-dev/android_device_zuk_z2_plus/issues/21#issuecomment-293812609

**Comment by poseidonss** [comment]
I had no more "white led of death" in this last build. I dont need to use wakelock anymore.

**Comment by AlvaroBrey** [comment]
@poseidonss @koichirose  That is probably because this last build has a constant kernel wakelock, sadly. We will see if the next release (without said wakelock) still works for you.

**Comment by koichirose** [comment]
@ontherunvaro an interesting thing: my problems where "fixed" approx on April 5, see [here](https://github.com/z2-dev/android_device_zuk_z2_plus/issues/21#issuecomment-293812609).
I moved to a new office in the beginning of April, I had zero black screens for months until a few days ago...when I moved back to the same area as my old office.
I got a couple black screens in the last few days. I'm pretty sure that, for me, this is related to some cell around where I work.

Just heard you'll stop supporting this ROM, I'm really sorry to hear that. Hope everything will be ok for you. Thank you for this amazing ROM, I wish you all the best.

**Comment by AlvaroBrey** [comment]
@koichirose We were suspecting modem from the start, weren't we? Your comment only makes me more suspicious of that.

Also, I will be ok, don't worry. Just going to be very busy for a few months.

**Comment by shaicoleman** [comment]
Workaround that works for me is to switch to 2G when charging and back to 3G on battery power. I use Automagic to automate that

**Comment by Mixxer5** [comment]
If someone's still looking at this thread/build: I wanted to confirm what @koichirose wrote. For me black screens started (again) with newest build (20171120) and wasn't present with older one (20171014). I updated my phone at work and it worked perfectly until I came home. Now it goes black right after reset (I can see launcher screen and then it goes off). So it must be some kind of network issue.

**Comment by sisqoboy** [comment]
Hi

Has any solution been found for this issue? Can we force the phone to reboot instead of the black screen?
